lib/msprime.c

```diff
@@ -4746,32 +4754,45 @@ msp_run_sweep(msp_t *self)
             rec_rates[j] = recomb_mass;
         }
 
-        curr_step++;
         event_prob = 1.0;
         event_rand = gsl_rng_uniform(self->rng);
         sweep_over = false;
         while (event_prob > event_rand && curr_step < num_steps && !sweep_over) {
             sweep_dt = time[curr_step] - time[curr_step - 1];
             /* using pop sizes grabbed from get_population_size */
             pop_size = get_population_size(&self->populations[0], time[curr_step]);
-            p_coal_B = ((sweep_pop_sizes[1] * (sweep_pop_sizes[1] - 1)))
-                       / allele_frequency[curr_step] * sweep_dt / pop_size;
-            p_coal_b = ((sweep_pop_sizes[0] * (sweep_pop_sizes[0] - 1)))
-                       / (1.0 - allele_frequency[curr_step]) * sweep_dt / pop_size;
+
+            // to avoid # lineages < 2
```

Comment:
I would structure this slightly differently to save a bit of vertical space @gbinux 

```c
p_coal_B = 0;
if (avl_count(&self->populations[0].ancestors[1]) > 1) {
     p_coal_B = ((sweep_pop_sizes[1] * (sweep_pop_sizes[1] - 1)))
￼                           / allele_frequency[curr_step] * sweep_dt / pop_size;
}
p_coal_b = 0;
...
```
I think the intention is very clear then and we don't need the comments.

Note as part of the project style we always use braces around indented blocks.